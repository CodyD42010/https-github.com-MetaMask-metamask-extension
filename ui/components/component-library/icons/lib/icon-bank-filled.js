import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconBankFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 399v62H47v-62c0-11 9-20 20-20h369c11 0 20 9 20 20zM149 236h-41v143h41zm82 0h-41v143h41zm82 0h-41v143h41zm82 0h-41v143h41zm82 240H26c-8 0-15-7-15-15 0-9 7-16 15-16h451c8 0 15 7 15 16 0 8-7 15-15 15zm-34-348L259 54c-4-1-11-1-15 0L60 128c-8 3-13 11-13 19v68c0 11 9 21 20 21h369c11 0 20-10 20-21v-68c0-8-5-16-13-19zm-191 56c-17 0-31-13-31-30s14-31 31-31 30 14 30 31-13 30-30 30z" />,
  );
};
IconBankFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
