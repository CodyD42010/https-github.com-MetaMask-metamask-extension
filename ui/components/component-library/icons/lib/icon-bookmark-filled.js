import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconBookmarkFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 170v172c0 66-53 119-119 119H166c-66 0-119-53-119-119V170c0-51 21-88 58-106 13-6 29 4 29 19v182c0 24 9 43 26 53s39 8 61-5l27-16c1-1 6-1 7-1l27 17c15 8 27 11 38 11 10 0 18-3 23-6 17-10 26-29 26-53V83c0-15 16-25 29-19 37 18 58 55 58 106zM318 51c11 0 21 9 21 21v193c0 13-4 23-11 26-7 5-18 2-30-5l-27-16c-11-6-28-6-39 0l-27 16c-12 7-23 9-30 5-7-3-11-13-11-26V72c0-12 10-21 21-21z" />,
  );
};
IconBookmarkFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
