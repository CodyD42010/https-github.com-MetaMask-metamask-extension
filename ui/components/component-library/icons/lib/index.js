export { IconAddSquareFilled } from './icon-add-square-filled';
export { IconBankFilled } from './icon-bank-filled';
export { IconBankTokenFilled } from './icon-bank-token-filled';
export { IconBookmarkFilled } from './icon-bookmark-filled';
export { IconCalculatorFilled } from './icon-calculator-filled';
export { IconCardFilled } from './icon-card-filled';
export { IconCardPosFilled } from './icon-card-pos-filled';
export { IconCardTokenFilled } from './icon-card-token-filled';
export { IconCategoryFilled } from './icon-category-filled';
export { IconChartFilled1 } from './icon-chart-filled-1';
export { IconChartFilled } from './icon-chart-filled';
export { IconCheckBoxOffOutline } from './icon-check-box-off-outline';
export { IconCheckBoxOnFilled } from './icon-check-box-on-filled';
export { IconCheckCircleOnFilled } from './icon-check-circle-on-filled';
export { IconClockFilled } from './icon-clock-filled';
export { IconCloseOutline } from './icon-close-outline';
export { IconCodeCircleFilled } from './icon-code-circle-filled';
export { IconCoinFilled } from './icon-coin-filled';
export { IconConnectFilled } from './icon-connect-filled';
export { IconCopyFilled } from './icon-copy-filled';
export { IconCopySuccessFilled } from './icon-copy-success-filled';
export { IconDangerFilled } from './icon-danger-filled';
export { IconDarkFilled } from './icon-dark-filled';
export { IconDataFilled } from './icon-data-filled';
export { IconDocumentCodeFilled } from './icon-document-code-filled';
export { IconEditFilled } from './icon-edit-filled';
export { IconEraserFilled } from './icon-eraser-filled';
export { IconExploreFilled } from './icon-explore-filled';
export { IconEyeFilled } from './icon-eye-filled';
export { IconEyeSlashFilled } from './icon-eye-slash-filled';
export { IconFilterOutline } from './icon-filter-outline';
export { IconFlagFilled } from './icon-flag-filled';
export { IconFlashFilled } from './icon-flash-filled';
export { IconFlashSlashFilled } from './icon-flash-slash-filled';
export { IconFullCircleFilled } from './icon-full-circle-filled';
export { IconGasFilled } from './icon-gas-filled';
export { IconGlobalFilled } from './icon-global-filled';
export { IconGlobalSearchFilled } from './icon-global-search-filled';
export { IconGraphFilled } from './icon-graph-filled';
export { IconHeartFilled } from './icon-heart-filled';
export { IconHierarchyFilled } from './icon-hierarchy-filled';
export { IconHomeFilled } from './icon-home-filled';
export { IconInfoFilled } from './icon-info-filled';
export { IconKeyFilled } from './icon-key-filled';
export { IconLightFilled } from './icon-light-filled';
export { IconLink2Outline } from './icon-link-2-outline';
export { IconLoadingFilled } from './icon-loading-filled';
export { IconLockCircleFilled } from './icon-lock-circle-filled';
export { IconLockFilled } from './icon-lock-filled';
export { IconLockSlashFilled } from './icon-lock-slash-filled';
export { IconMenuOutline } from './icon-menu-outline';
export { IconMessageQuestionFilled } from './icon-message-question-filled';
export { IconMessagesFilled } from './icon-messages-filled';
export { IconMinusSquareFilled } from './icon-minus-square-filled';
export { IconMobileFilled } from './icon-mobile-filled';
export { IconMoneyFilled } from './icon-money-filled';
export { IconMonitorFilled } from './icon-monitor-filled';
export { IconMoreHorizontalOutline } from './icon-more-horizontal-outline';
export { IconMoreVerticalOutline } from './icon-more-vertical-outline';
export { IconNotificationCircleFilled } from './icon-notification-circle-filled';
export { IconNotificationFilled } from './icon-notification-filled';
export { IconPasswordCheckFilled } from './icon-password-check-filled';
export { IconPeopleFilled } from './icon-people-filled';
export { IconProgrammingArrowsFilled } from './icon-programming-arrows-filled';
export { IconQuestionFilled } from './icon-question-filled';
export { IconReceivedOutline } from './icon-received-outline';
export { IconScanBarcodeFilled } from './icon-scan-barcode-filled';
export { IconScanFilled } from './icon-scan-filled';
export { IconScrollFilled } from './icon-scroll-filled';
export { IconSearchFilled } from './icon-search-filled';
export { IconSecurityCardFilled } from './icon-security-card-filled';
export { IconSecurityCrossFilled } from './icon-security-cross-filled';
export { IconSecurityKeyFilled } from './icon-security-key-filled';
export { IconSecuritySearchFilled } from './icon-security-search-filled';
export { IconSecuritySlashFilled } from './icon-security-slash-filled';
export { IconSecurityTickFilled } from './icon-security-tick-filled';
export { IconSecurityTimeFilled } from './icon-security-time-filled';
export { IconSecurityUserFilled } from './icon-security-user-filled';
export { IconSend1Filled } from './icon-send-1-filled';
export { IconSend2Outline } from './icon-send-2-outline';
export { IconSettingFilled } from './icon-setting-filled';
export { IconSlashFilled } from './icon-slash-filled';
export { IconSnapsFilled } from './icon-snaps-filled';
export { IconSnapsMobileFilled } from './icon-snaps-mobile-filled';
export { IconSnapsPlusFilled } from './icon-snaps-plus-filled';
export { IconSpeedometerFilled } from './icon-speedometer-filled';
export { IconStudentFilled } from './icon-student-filled';
export { IconSwapHorizontalOutline } from './icon-swap-horizontal-outline';
export { IconSwapVerticalFilled } from './icon-swap-vertical-filled';
export { IconTagFilled } from './icon-tag-filled';
export { IconTimerFilled } from './icon-timer-filled';
export { IconTrashFilled } from './icon-trash-filled';
export { IconTrendDownFilled } from './icon-trend-down-filled';
export { IconTrendUpFilled } from './icon-trend-up-filled';
export { IconUserAddFilled1 } from './icon-user-add-filled-1';
export { IconUserAddFilled } from './icon-user-add-filled';
export { IconUserCircleAddFilled } from './icon-user-circle-add-filled';
export { IconUserCircleFilled } from './icon-user-circle-filled';
export { IconUserFilled } from './icon-user-filled';
export { IconUserMinusFilled } from './icon-user-minus-filled';
export { IconUserRemoveFilled } from './icon-user-remove-filled';
export { IconUserSearchFilled } from './icon-user-search-filled';
export { IconUserTickFilled } from './icon-user-tick-filled';
export { IconWalletCardFilled } from './icon-wallet-card-filled';
export { IconWalletFilled } from './icon-wallet-filled';
export { IconWalletMoneyFilled } from './icon-wallet-money-filled';
export { IconWarningFilled } from './icon-warning-filled';
