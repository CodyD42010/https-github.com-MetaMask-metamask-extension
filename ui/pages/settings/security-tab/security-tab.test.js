import { fireEvent, queryByRole, screen } from '@testing-library/react';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import mockState from '../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import SecurityTab from './security-tab.container';

jest.mock('../../../../app/scripts/lib/util', () => {
  const originalModule = jest.requireActual('../../../../app/scripts/lib/util');

  return {
    ...originalModule,
    getEnvironmentType: jest.fn(),
  };
});

describe('Security Tab', () => {
  delete mockState.metamask.featureFlags; // Unset featureFlags in order to test the default value
  mockState.appState.warning = 'warning'; // This tests an otherwise untested render branch

  const mockStore = configureMockStore([thunk])(mockState);

  function toggleCheckbox(testId, initialState) {
    renderWithProvider(<SecurityTab />, mockStore);

    const container = screen.getByTestId(testId);
    const checkbox = queryByRole(container, 'checkbox');

    expect(checkbox).toHaveAttribute('value', initialState ? 'true' : 'false');

    fireEvent.click(checkbox); // This fires the onToggle method of the ToggleButton, but it doesn't change the value of the checkbox

    fireEvent.change(checkbox, {
      target: { value: !initialState }, // This changes the value of the checkbox
    });

    expect(checkbox).toHaveAttribute('value', initialState ? 'false' : 'true');

    return true;
  }

  it('should match snapshot', () => {
    const { container } = renderWithProvider(<SecurityTab />, mockStore);

    expect(container).toMatchSnapshot();
  });

  it('toggles phishing detection', async () => {
    expect(await toggleCheckbox('usePhishingDetection', true)).toBe(true);
  });

  it('toggles balance and token price checker', async () => {
    expect(await toggleCheckbox('currencyRateCheckToggle', true)).toBe(true);
  });

  it('toggles incoming txs', async () => {
    expect(await toggleCheckbox('showIncomingTransactions', false)).toBe(true);
  });

  it('should toggle token detection', async () => {
    expect(await toggleCheckbox('autoDetectTokens', true)).toBe(true);
  });

  it('toggles batch balance checks', async () => {
    expect(await toggleCheckbox('useMultiAccountBalanceChecker', false)).toBe(
      true,
    );
  });

  it('toggles metaMetrics', async () => {
    expect(await toggleCheckbox('participateInMetaMetrics', false)).toBe(true);
  });

  it('toggles SRP Quiz', async () => {
    renderWithProvider(<SecurityTab />, mockStore);

    expect(
      screen.queryByTestId(`srp_stage_introduction`),
    ).not.toBeInTheDocument();

    fireEvent.click(screen.getByTestId('reveal-seed-words'));

    expect(screen.getByTestId(`srp_stage_introduction`)).toBeInTheDocument();

    const container = screen.getByTestId('srp-quiz-header');
    const checkbox = queryByRole(container, 'button');
    fireEvent.click(checkbox);

    expect(
      screen.queryByTestId(`srp_stage_introduction`),
    ).not.toBeInTheDocument();
  });
});
