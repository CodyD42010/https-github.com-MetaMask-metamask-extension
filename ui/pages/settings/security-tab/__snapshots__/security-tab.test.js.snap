// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security Tab should match snapshot 1`] = `
<div>
  <div
    class="settings-page__body"
  >
    <div
      class="settings-tab__error"
    >
      warning
    </div>
    <span
      class="settings-page__security-tab-sub-header__bold"
    >
      Security
    </span>
    <div
      class="settings-page__security-tab-sub-header"
    >
      Secret Recovery Phrase
    </div>
    <div
      class="settings-page__content-padded"
    >
      <button
        class="box mm-text mm-button-base mm-button-base--size-lg mm-button-primary mm-text--body-md box--padding-right-4 box--padding-left-4 box--display-inline-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-primary-inverse box--background-color-primary-default box--rounded-pill"
        data-testid="reveal-seed-words"
        type="danger"
      >
        Reveal Secret Recovery Phrase
      </button>
    </div>
    <span
      class="settings-page__security-tab-sub-header__bold"
    >
      Privacy
    </span>
    <div>
      <span
        class="settings-page__security-tab-sub-header"
      >
        Alerts
      </span>
    </div>
    <div
      class="settings-page__content-padded"
    >
      <div
        class="settings-page__content-row"
      >
        <div
          class="settings-page__content-item"
        >
          <span>
            Use phishing detection
          </span>
          <div
            class="settings-page__content-description"
          >
            Display a warning for phishing domains targeting Ethereum users
          </div>
        </div>
        <div
          class="settings-page__content-item"
        >
          <div
            class="settings-page__content-item-col"
            data-testid="usePhishingDetection"
          >
            <label
              class="toggle-button toggle-button--on"
              tabindex="0"
            >
              <div
                style="display: flex; width: 52px; align-items: center; justify-content: flex-start; position: relative; cursor: pointer; background-color: transparent; border: 0px; padding: 0px; user-select: none;"
              >
                <div
                  style="width: 40px; height: 24px; padding: 0px; border-radius: 26px; display: flex; align-items: center; justify-content: center; background-color: rgb(242, 244, 246);"
                >
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgb(250, 250, 250); margin-top: auto; margin-bottom: auto; line-height: 0; opacity: 1; width: 26px; height: 20px; left: 4px;"
                  />
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgba(255, 255, 255, 0.6); bottom: 0px; margin-top: auto; margin-bottom: auto; padding-right: 5px; line-height: 0; width: 26px; height: 20px; opacity: 0;"
                  />
                </div>
                <div
                  style="position: absolute; height: 100%; top: 0px; left: 0px; display: flex; flex: 1; align-self: stretch; align-items: center; justify-content: flex-start;"
                >
                  <div
                    style="width: 18px; height: 18px; display: flex; align-self: center; box-shadow: none; border-radius: 50%; box-sizing: border-box; position: relative; background-color: rgb(3, 125, 214); left: 18px;"
                  />
                </div>
                <input
                  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
                  type="checkbox"
                  value="true"
                />
              </div>
              <div
                class="toggle-button__status"
              >
                <span
                  class="toggle-button__label-off"
                >
                  Off
                </span>
                <span
                  class="toggle-button__label-on"
                >
                  On
                </span>
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
    <span
      class="settings-page__security-tab-sub-header"
    >
      Transactions
    </span>
    <div
      class="settings-page__content-padded"
    >
      <div
        class="settings-page__content-row"
      >
        <div
          class="settings-page__content-item"
        >
          <span>
            Show balance and token price checker
          </span>
          <div
            class="settings-page__content-description"
          >
            <span>
               
              We use 
              <a
                href="https://www.coingecko.com/"
                rel="noreferrer"
                target="_blank"
              >
                CoinGecko
              </a>
               and 
              <a
                href="https://www.cryptocompare.com/"
                rel="noreferrer"
                target="_blank"
              >
                CryptoCompare
              </a>
               APIs to display your balance and token price. 
              <a
                href="https://consensys.net/privacy-policy/"
                rel="noreferrer"
                target="_blank"
              >
                Privacy policy
              </a>
              
               
            </span>
          </div>
        </div>
        <div
          class="settings-page__content-item"
        >
          <div
            class="settings-page__content-item-col"
            data-testid="currencyRateCheckToggle"
          >
            <label
              class="toggle-button toggle-button--on"
              tabindex="0"
            >
              <div
                style="display: flex; width: 52px; align-items: center; justify-content: flex-start; position: relative; cursor: pointer; background-color: transparent; border: 0px; padding: 0px; user-select: none;"
              >
                <div
                  style="width: 40px; height: 24px; padding: 0px; border-radius: 26px; display: flex; align-items: center; justify-content: center; background-color: rgb(242, 244, 246);"
                >
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgb(250, 250, 250); margin-top: auto; margin-bottom: auto; line-height: 0; opacity: 1; width: 26px; height: 20px; left: 4px;"
                  />
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgba(255, 255, 255, 0.6); bottom: 0px; margin-top: auto; margin-bottom: auto; padding-right: 5px; line-height: 0; width: 26px; height: 20px; opacity: 0;"
                  />
                </div>
                <div
                  style="position: absolute; height: 100%; top: 0px; left: 0px; display: flex; flex: 1; align-self: stretch; align-items: center; justify-content: flex-start;"
                >
                  <div
                    style="width: 18px; height: 18px; display: flex; align-self: center; box-shadow: none; border-radius: 50%; box-sizing: border-box; position: relative; background-color: rgb(3, 125, 214); left: 18px;"
                  />
                </div>
                <input
                  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
                  type="checkbox"
                  value="true"
                />
              </div>
              <div
                class="toggle-button__status"
              >
                <span
                  class="toggle-button__label-off"
                >
                  Off
                </span>
                <span
                  class="toggle-button__label-on"
                >
                  On
                </span>
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="settings-page__content-row"
      >
        <div
          class="settings-page__content-item"
        >
          <span>
            Show incoming transactions
          </span>
          <div
            class="settings-page__content-description"
          >
            <span>
               
              This relies on 
              <a
                href="https://etherscan.io/privacyPolicy"
                rel="noopener noreferrer"
                target="_blank"
              >
                Etherscan
              </a>
               which will have access to your Ethereum address and your IP address. 
              <a
                href="https://consensys.net/privacy-policy/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Privacy policy
              </a>
              
               
            </span>
          </div>
        </div>
        <div
          class="settings-page__content-item"
        >
          <div
            class="settings-page__content-item-col"
            data-testid="showIncomingTransactions"
          >
            <label
              class="toggle-button toggle-button--off"
              tabindex="0"
            >
              <div
                style="display: flex; width: 52px; align-items: center; justify-content: flex-start; position: relative; cursor: pointer; background-color: transparent; border: 0px; padding: 0px; user-select: none;"
              >
                <div
                  style="width: 40px; height: 24px; padding: 0px; border-radius: 26px; display: flex; align-items: center; justify-content: center; background-color: rgb(242, 244, 246);"
                >
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgb(250, 250, 250); margin-top: auto; margin-bottom: auto; line-height: 0; opacity: 0; width: 26px; height: 20px; left: 4px;"
                  />
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgba(255, 255, 255, 0.6); bottom: 0px; margin-top: auto; margin-bottom: auto; padding-right: 5px; line-height: 0; width: 26px; height: 20px; opacity: 1;"
                  />
                </div>
                <div
                  style="position: absolute; height: 100%; top: 0px; left: 0px; display: flex; flex: 1; align-self: stretch; align-items: center; justify-content: flex-start;"
                >
                  <div
                    style="width: 18px; height: 18px; display: flex; align-self: center; box-shadow: none; border-radius: 50%; box-sizing: border-box; position: relative; background-color: rgb(106, 115, 125); left: 3px;"
                  />
                </div>
                <input
                  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
                  type="checkbox"
                  value="false"
                />
              </div>
              <div
                class="toggle-button__status"
              >
                <span
                  class="toggle-button__label-off"
                >
                  Off
                </span>
                <span
                  class="toggle-button__label-on"
                >
                  On
                </span>
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
    <span
      class="settings-page__security-tab-sub-header"
    >
      Network provider
    </span>
    <div
      class="settings-page__content-padded"
    >
      <div
        class="settings-page__content-row"
        data-testid="advanced-setting-choose-your-network"
      >
        <div
          class="settings-page__content-item"
        >
          <span>
            Choose your network
          </span>
          <div
            class="settings-page__content-description"
          >
            <span>
               
              We use Infura as our remote procedure call (RPC) provider to offer the most reliable and private access to Ethereum data we can. You can choose your own RPC, but remember that any RPC will receive your IP address and Ethereum wallet to make transactions. Read our 
              <a
                href="https://consensys.net/privacy-policy/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Privacy policy
              </a>
               to learn more about how Infura handles data.
               
            </span>
          </div>
        </div>
        <div
          class="settings-page__content-item"
        >
          <div
            class="settings-page__content-item-col"
          >
            <button
              class="box mm-text mm-button-base mm-button-base--size-md settings-page__button mm-button-primary mm-text--body-md box--padding-right-4 box--padding-left-4 box--display-inline-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-primary-inverse box--background-color-primary-default box--rounded-pill"
              type="secondary"
            >
              Add custom network
            </button>
          </div>
        </div>
      </div>
      <div
        class="settings-page__content-row"
        data-testid="setting-ipfs-gateway"
      >
        <div
          class="settings-page__content-item"
        >
          <span>
            Add custom IPFS gateway
          </span>
          <div
            class="settings-page__content-description"
          >
            The IPFS gateway makes it possible to access and view data hosted by third parties. You can add a custom IPFS gateway or continue using the default.
          </div>
        </div>
        <div
          class="settings-page__content-item"
        >
          <div
            class="settings-page__content-item-col"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root TextField-inputRoot-12 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  dir="auto"
                  type="text"
                  value="dweb.link"
                />
              </div>
              
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      class="settings-page__security-tab-sub-header"
    >
      Token autodetection
    </span>
    <div
      class="settings-page__content-padded"
    >
      <div
        class="settings-page__content-row"
        data-testid="advanced-setting-gas-fee-estimation"
      >
        <div
          class="settings-page__content-item"
        >
          <span>
            Autodetect tokens
          </span>
          <div
            class="settings-page__content-description"
          >
            <span>
               
              We use third-party APIs to detect and display new tokens sent to your wallet. Turn off if you don’t want the app to automatically pull data from those services. 
              <a
                href="https://consensys.net/privacy-policy/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Learn More
              </a>
              
               
            </span>
          </div>
        </div>
        <div
          class="settings-page__content-item"
        >
          <div
            class="settings-page__content-item-col"
            data-testid="autoDetectTokens"
          >
            <label
              class="toggle-button toggle-button--on"
              tabindex="0"
            >
              <div
                style="display: flex; width: 52px; align-items: center; justify-content: flex-start; position: relative; cursor: pointer; background-color: transparent; border: 0px; padding: 0px; user-select: none;"
              >
                <div
                  style="width: 40px; height: 24px; padding: 0px; border-radius: 26px; display: flex; align-items: center; justify-content: center; background-color: rgb(242, 244, 246);"
                >
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgb(250, 250, 250); margin-top: auto; margin-bottom: auto; line-height: 0; opacity: 1; width: 26px; height: 20px; left: 4px;"
                  />
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgba(255, 255, 255, 0.6); bottom: 0px; margin-top: auto; margin-bottom: auto; padding-right: 5px; line-height: 0; width: 26px; height: 20px; opacity: 0;"
                  />
                </div>
                <div
                  style="position: absolute; height: 100%; top: 0px; left: 0px; display: flex; flex: 1; align-self: stretch; align-items: center; justify-content: flex-start;"
                >
                  <div
                    style="width: 18px; height: 18px; display: flex; align-self: center; box-shadow: none; border-radius: 50%; box-sizing: border-box; position: relative; background-color: rgb(3, 125, 214); left: 18px;"
                  />
                </div>
                <input
                  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
                  type="checkbox"
                  value="true"
                />
              </div>
              <div
                class="toggle-button__status"
              >
                <span
                  class="toggle-button__label-off"
                >
                  Off
                </span>
                <span
                  class="toggle-button__label-on"
                >
                  On
                </span>
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="settings-page__content-row"
      >
        <div
          class="settings-page__content-item"
        >
          <span>
            Batch account balance requests
          </span>
          <div
            class="settings-page__content-description"
          >
            We batch accounts and query Infura to responsively show your balances. If you turn this off, only active accounts will be queried. Some dapps won't work unless you connect your wallet.
          </div>
        </div>
        <div
          class="settings-page__content-item"
        >
          <div
            class="settings-page__content-item-col"
            data-testid="useMultiAccountBalanceChecker"
          >
            <label
              class="toggle-button toggle-button--off"
              tabindex="0"
            >
              <div
                style="display: flex; width: 52px; align-items: center; justify-content: flex-start; position: relative; cursor: pointer; background-color: transparent; border: 0px; padding: 0px; user-select: none;"
              >
                <div
                  style="width: 40px; height: 24px; padding: 0px; border-radius: 26px; display: flex; align-items: center; justify-content: center; background-color: rgb(242, 244, 246);"
                >
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgb(250, 250, 250); margin-top: auto; margin-bottom: auto; line-height: 0; opacity: 0; width: 26px; height: 20px; left: 4px;"
                  />
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgba(255, 255, 255, 0.6); bottom: 0px; margin-top: auto; margin-bottom: auto; padding-right: 5px; line-height: 0; width: 26px; height: 20px; opacity: 1;"
                  />
                </div>
                <div
                  style="position: absolute; height: 100%; top: 0px; left: 0px; display: flex; flex: 1; align-self: stretch; align-items: center; justify-content: flex-start;"
                >
                  <div
                    style="width: 18px; height: 18px; display: flex; align-self: center; box-shadow: none; border-radius: 50%; box-sizing: border-box; position: relative; background-color: rgb(106, 115, 125); left: 3px;"
                  />
                </div>
                <input
                  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
                  type="checkbox"
                  value="false"
                />
              </div>
              <div
                class="toggle-button__status"
              >
                <span
                  class="toggle-button__label-off"
                >
                  Off
                </span>
                <span
                  class="toggle-button__label-on"
                >
                  On
                </span>
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
    <span
      class="settings-page__security-tab-sub-header"
    >
      Metrics
    </span>
    <div
      class="settings-page__content-padded"
    >
      <div
        class="settings-page__content-row"
      >
        <div
          class="settings-page__content-item"
        >
          <span>
            Participate in MetaMetrics
          </span>
          <div
            class="settings-page__content-description"
          >
            <span>
              Participate in MetaMetrics to help us make MetaMask better
            </span>
          </div>
        </div>
        <div
          class="settings-page__content-item"
        >
          <div
            class="settings-page__content-item-col"
            data-testid="participateInMetaMetrics"
          >
            <label
              class="toggle-button toggle-button--off"
              tabindex="0"
            >
              <div
                style="display: flex; width: 52px; align-items: center; justify-content: flex-start; position: relative; cursor: pointer; background-color: transparent; border: 0px; padding: 0px; user-select: none;"
              >
                <div
                  style="width: 40px; height: 24px; padding: 0px; border-radius: 26px; display: flex; align-items: center; justify-content: center; background-color: rgb(242, 244, 246);"
                >
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgb(250, 250, 250); margin-top: auto; margin-bottom: auto; line-height: 0; opacity: 0; width: 26px; height: 20px; left: 4px;"
                  />
                  <div
                    style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgba(255, 255, 255, 0.6); bottom: 0px; margin-top: auto; margin-bottom: auto; padding-right: 5px; line-height: 0; width: 26px; height: 20px; opacity: 1;"
                  />
                </div>
                <div
                  style="position: absolute; height: 100%; top: 0px; left: 0px; display: flex; flex: 1; align-self: stretch; align-items: center; justify-content: flex-start;"
                >
                  <div
                    style="width: 18px; height: 18px; display: flex; align-self: center; box-shadow: none; border-radius: 50%; box-sizing: border-box; position: relative; background-color: rgb(106, 115, 125); left: 3px;"
                  />
                </div>
                <input
                  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
                  type="checkbox"
                  value="false"
                />
              </div>
              <div
                class="toggle-button__status"
              >
                <span
                  class="toggle-button__label-off"
                >
                  Off
                </span>
                <span
                  class="toggle-button__label-on"
                >
                  On
                </span>
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
